### PR TITLE
Adding Taylor & Francis Journal Template

### PR DESCRIPTION
**Pull request by dleutnant**
To contribute a new article template to this package, please make sure you have done the following things (note that `journalname_article` below is only an example name):

- [x] Unless you have done it in any other RStudio's projects before, please sign the [individual](https://rstudioblog.files.wordpress.com/2017/05/rstudio_individual_contributor_agreement.pdf) or [corporate](https://rstudioblog.files.wordpress.com/2017/05/rstudio_corporate_contributor_agreement.pdf) contributor agreement for a significant pull request (it is fine not to sign it if a PR is only intended to fix a few typos). You can send the signed copy to <jj@rstudio.com>.

- [x] Add the `journalname_article()` function to `R/article.R` if the output format is simple enough, otherwise create a separate `R/journalname_article.R`.

- [x] Add the Pandoc LaTeX template `inst/rmarkdown/templates/journalname_article/resources/template.tex`.

- [x] Add a skeleton article `inst/rmarkdown/templates/journalname_article/skeleton/skeleton.Rmd`.

- [x] Add a description of the template `inst/rmarkdown/templates/journalname_article/template.yaml`.

- [x] Please include the document class file (`*.cls`) if needed, but please do not include standard LaTeX packages (`*.sty`) that can be downloaded from CTAN. Please keep the number of new files absolutely minimal, and also make examples minimal (e.g., if you need a `.bib` example, try to only leave one or two bibliography entries in it, and don't include one hundred items in it without using all of them).

- [x] Update Rd and namespace (could be done by `devtools::document()`).

- [x] Update NEWS.

- [x] Update README with a link to the newly supported journal.

- [x] Add a test to `tests/testit/test-formats.R`.

- [x] Add your name to the list of authors `Authors@R` in DESCRIPTION. You don't need to bump the package version in DESCRIPTION.

Lastly, please try your best to do only one thing per pull request (e.g., if you want to add two output formats, do them in two separate pull requests), and refrain from making cosmetic changes in the code base: https://yihui.name/en/2018/02/bite-sized-pull-requests/

Thank you!

### CODE REVIEW

**Comment by dleutnant** [comment]
@yihui Are you going to resolve the conflicts or is it me?

**Comment by yihui** [comment]
It seems you have resolved conflicts by yourself. Good job! I'll review the PR later.

File: R/article.R [hunk 1]
```diff
@@ -261,3 +261,13 @@ springer_article <- function(..., keep_tex = TRUE, citation_package = 'none'){
     "springer_article", keep_tex = keep_tex, citation_package = citation_package, ...
   )
 }
+
+#' @section \code{tf_article}: Format for creating submissions to a Taylor & Francis journal. Adapted from
+#' \href{http://www.tandf.co.uk/journals/authors/InteractCADLaTeX.zip}{http://www.tandf.co.uk/journals/authors/InteractCADLaTeX.zip}.
```
Comment (yihui) [inline R/article.R]:
```suggestion
#' \url{http://www.tandf.co.uk/journals/authors/InteractCADLaTeX.zip}.
```

File: R/article.R [hunk 2]
```diff
@@ -261,3 +261,13 @@ springer_article <- function(..., keep_tex = TRUE, citation_package = 'none'){
...
+#' @export
+#' @rdname article
+tf_article <- function(..., keep_tex = TRUE, citation_package = 'natbib') {
+  pdf_document_format(
+    "tf_article", keep_tex = keep_tex, citation_package = "natbib", ...
```
Comment (yihui) [inline R/article.R]:
```suggestion
    "tf_article", keep_tex = keep_tex, citation_package = citation_package, ...
```

File: inst/rmarkdown/templates/tf_article/skeleton/skeleton.Rmd [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+header-includes: |
+  \usepackage{hyperref}
+  \usepackage[utf8]{inputenc}
+  \def\tightlist{}
+output: rticles::tf_article
+---
+
+# Introduction
+
+In order to assist authors in the process of preparing a manuscript for a journal, the Taylor & Francis `\textsf{Interact}' layout style has been implemented as a \LaTeXe\ class file based on the \texttt{article} document class. A \textsc{Bib}\TeX\ bibliography style file and a sample bibliography are also provided in order to assist with the formatting of your references.
```
Comment (yihui) [inline inst/rmarkdown/templates/tf_article/skeleton/skeleton.Rmd]: I think the backtick before `\textsf` should be a single quote.
```suggestion
In order to assist authors in the process of preparing a manuscript for a journal, the Taylor & Francis '\textsf{Interact}' layout style has been implemented as a \LaTeXe\ class file based on the \texttt{article} document class. A \textsc{Bib}\TeX\ bibliography style file and a sample bibliography are also provided in order to assist with the formatting of your references.
```

**Comment by dleutnant** [comment]
I substituted a couple of raw LaTeX codes. I didn't change the way figures and tables are integrated and would like the end user to take care of this. Equations are still enveloped in a LaTeX environment to enable enumeration of equations.

File: R/article.R [hunk 4]
```diff
@@ -261,3 +261,13 @@ springer_article <- function(..., keep_tex = TRUE, citation_package = 'none'){
...
 }
+
+#' @section \code{tf_article}: Format for creating submissions to a Taylor & Francis journal. Adapted from
+#' \url{http://www.tandf.co.uk/journals/authors/InteractCADLaTeX.zip}.
+#' @export
+#' @rdname article
+tf_article <- function(..., keep_tex = TRUE, citation_package = 'natbib') {
+  pdf_document_format(
+    "tf_article", keep_tex = keep_tex, citation_package = citation_package, ...
+    )
```
Comment (yihui) [inline R/article.R]:
```suggestion
  )
```

File: inst/rmarkdown/templates/tf_article/skeleton/skeleton.Rmd [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+* `oneside`, `twoside` -- not necessary, `oneside` is the default;
+* `leqno`, `titlepage` -- should not be used;
+* `twocolumn` -- should not be used (see Subsection~\ref{class});
+* `onecolumn` -- not necessary as it is the default style.
+
+To prepare a manuscript for a journal that is printed in A4 (two column) format, use the `largeformat` document-class option provided by `interact.cls`; otherwise the class file produces pages sized for B5 (single column) format by default. The `geometry` package should not be used to make any further adjustments to the page dimensions.
+
+<!-- If your manuscript has supplementary content you can prepare this using the `suppldata` document-class option, which will suppress the `article history' date. This option must _not_ be used on any primary content. -->
+
+# Additional features of the `interact} class file
```
Comment (yihui) [inline inst/rmarkdown/templates/tf_article/skeleton/skeleton.Rmd]:
```suggestion
# Additional features of the `interact` class file
```

**Comment by dleutnant** [comment]
Thank you!

**Pull request merged**